src/main/java/com/google/codeu/servlets/MessageServlet.java

```diff
@@ -129,30 +134,44 @@ private String getUploadedFileUrl(HttpServletRequest request, String formInputEl
       return null;
     }
 
-    // Our form only contains a single file input, so get the first index.
-    BlobKey blobKey = blobKeys.get(0);
-
     // User submitted form without selecting a file, so we can't get a URL. (live server)
-    BlobInfo blobInfo = new BlobInfoFactory().loadBlobInfo(blobKey);
-    if (blobInfo.getSize() == 0) {
-      blobstoreService.delete(blobKey);
-      return null;
+    for(BlobKey blobKey: blobKeys)
+    {
+      BlobInfo blobInfo = new BlobInfoFactory().loadBlobInfo(blobKey);
+      if (blobInfo.getSize() == 0) {
+        blobstoreService.delete(blobKey);
+      }
     }
-
+	
     // Use ImagesService to get a URL that points to the uploaded file.
     ImagesService imagesService = ImagesServiceFactory.getImagesService();
-    ServingUrlOptions options = ServingUrlOptions.Builder.withBlobKey(blobKey);  
-    // Checking the validity of the file to make sure it's an image, if not catch exception.
-    try
-    {
-        //getting the image URL to the uploaded file
-        String imageUrl = imagesService.getServingUrl(options);
-        return imageUrl;
-    }
-    catch(java.lang.IllegalArgumentException exception)
+    List<String> imageBlobUrls = new ArrayList<String>();
+	
+    for(BlobKey blobK: blobKeys)
     {
-        return null;
+      // Checking the validity of the file to make sure it's an image, if not catch exception.
+      String fileType = new BlobInfoFactory().loadBlobInfo(blobK).getContentType().toString().toLowerCase();
+      if(!(fileType.equals("image/jpg") ||fileType.equals("image/jpeg") || fileType.equals("image/gif")))
+      {
+        blobstoreService.delete(blobK);
+      }
+	  else //else: blob is an image
+      {
+        try
+        {
+          //getting the image URL to the uploaded file
+          ServingUrlOptions options = ServingUrlOptions.Builder.withBlobKey(blobK);  
+          String imageUrl = imagesService.getServingUrl(options);//getServingUrl locks the blob, so it cannot be deleted with blobstoreService.delete(blobK); 
+          imageBlobUrls.add(imageUrl);
+	}
+	catch(IllegalArgumentException e)
+	{
+          //in case getServingUrl() raises an exception	
```

Comment:
Do you think we can possibly console.log the exception to make debugging easier?

Reply:
This exception catch phrase is there if a file type is not image, but because I check the file type before it gets to try/ catch block, there is almost no chance of it happening. I wanted to delete it, but then I thought maybe I should just leave it with an empty body.

Reply:
Gotcha, sounds good. Can you just make that a comment then in case somebody who is reading our code has the same thought?

Reply:
Yes sure

Reply:
fix it! Thanks for letting me know about this